File: 872-leaf-similar-trees/872-leaf-similar-trees.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def traversal(self,root,l):
        if(not root): return
        if(not root.left and not root.right): l.append(root.val)
        if(root.left): self.traversal(root.left,l)
        if(root.right): self.traversal(root.right,l)
    
    def leafSimilar(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> bool:
        a,b=[],[]
        self.traversal(root1,a)
        self.traversal(root2,b)
        return a==b

File: 872-leaf-similar-trees/872-leaf-similar-trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void traversal(TreeNode* root, vector<int>& v){
        if(!root) return;
        if(!root->left && !root->right) v.push_back(root->val);
        if(root->left) traversal(root->left,v);
        if(root->right) traversal(root->right,v);
    }
    
    bool leafSimilar(TreeNode* root1, TreeNode* root2) {
        vector<int> a,b;
        traversal(root1,a);
        traversal(root2,b);
        return a==b;
    }
};